src/Garn/Optparse.hs

```diff
@@ -25,19 +28,32 @@ getOpts oType =
   where
     unavailable :: OA.Doc
     unavailable =
-      let formatCommands cmdInfo = [PP.string cmd | (cmd, _, _) <- cmdInfo]
+      let fst3 (cmd, _, _) = cmd
+          formatCommands = fmap PP.string
        in PP.nest
             2
             $ PP.vsep
               ( PP.string "Unavailable commands:"
                   : case oType of
-                    WithGarnTs _ -> formatCommands withouGarnTsCommandInfo
-                    WithoutGarnTs -> formatCommands withGarnTsCommandInfo
+                    WithGarnTs _ -> formatCommands $ fst3 <$> withoutGarnTsCommandInfo
+                    WithoutGarnTs -> formatCommands $ fst3 <$> withGarnTsCommandInfo
+                    BrokenGarnTs _ ->
+                      formatCommands
+                        ( (fst3 <$> withGarnTsCommandInfo)
+                            ++ (fst3 <$> withoutGarnTsCommandInfo)
+                        )
               )
     parser :: Parser Options
     parser = case oType of
-      WithGarnTs garnConfig -> WithGarnTsOpts garnConfig <$> withGarnTsParser (targets garnConfig)
-      WithoutGarnTs -> WithoutGarnTsOpts <$> withoutGarnTsParser
+      WithGarnTs garnConfig ->
+        (WithGarnTsOpts garnConfig <$> withGarnTsParser (targets garnConfig))
+          <|> (AlwaysAvailableOpts <$> alwaysParser Hidden)
+      WithoutGarnTs ->
+        (WithoutGarnTsOpts <$> withoutGarnTsParser)
+          <|> (AlwaysAvailableOpts <$> alwaysParser Hidden)
+      BrokenGarnTs err ->
+        (AlwaysAvailableOpts <$> alwaysParser Hidden)
+          <|> throw (Garn.Errors.UserError $ errorMessage err)
```

Comment:
Regarding throwing a pure exception here and tab-completion: Currently syntax errors in the garn file allow to run `garn edit`, but tab-completion doesn't work nicely, it just outputs the garn deno error. Here's a test that fails and proves that:

```haskell
        it "tab-completes edit when garn.ts has syntax error" $ \onTestFailureLog -> do
          writeFile "garn.ts" "{ syntax err"
          output <- runGarn ["--bash-completion-index", "0"] "" repoDir Nothing
          onTestFailureLog output
          stdout output
            `shouldBe` unindent
              [i|
                edit
                -h
                --help
                --version
              |]
```

I didn't find a very easy way to avoid that. Turning the main parser into a `Parser (Either UserError Options)` fixes the issues with tab-completion, but means that when running other commands (e.g. `garn run`), it'll *not* output the deno error, but instead a cli help message saying that `run` is unavailable, but `edit` is allowed.

So -- unless we find a solution quickly -- I'm inclined to merge this as is, since this PR doesn't make things worse.